libvast/src/systemd.cpp

```diff
@@ -15,6 +17,27 @@ namespace vast::systemd {
 
 #if VAST_LINUX
 
+bool connected_to_journal() {
+  auto journal_env = detail::locked_getenv("JOURNAL_STREAM");
+  if (!journal_env)
+    return false;
+  size_t device_number = 0;
+  char colon = 0;
+  size_t inode_number = 0;
+  auto parser = parsers::u64 >> parsers::ch<':'> >> parsers::u64;
+  if (parser(*journal_env, device_number, colon, inode_number)) {
+    VAST_WARN("could not parse systemd environment variable "
+              "$JOURNAL_STREAM='{}'",
+              *journal_env);
+    return false;
+  }
+  struct stat buf = {};
+  // TODO: errno-ify this
+  auto stderrfd = ::fileno(stderr);
+  ::fstat(stderrfd, &buf);
+  return buf.st_dev == device_number && buf.st_ino == inode_number;
```

Comment:
Is this doable with `std::filesystem` instead?

Reply:
I highly doubt it, `std::filesystem` is too high-level to know about device or inode numbers.